### PR TITLE
Add label smoothing feature in sampled softmax

### PR DESCRIPTION
**Pull request by wan-wei**

### CODE REVIEW

**Comment by drpngx** [comment]
We need a test for that.

**Comment by drpngx** [comment]
So, I think what needs to happen is that the smoothing is only applied to the other targets, so the probability is divided by `num_classes - 1`. For the one-hot, you can use the `1 - p` as `on_value`, and `p / (n -1)` as `off_value`.

**Comment by nestle1993** [comment]
@drpngx  Thanks for your advice!

Indeed, I got inf cost when I tested this function on my language model, which has about 80,000 classes in total, with `label_smoothing=0.2`, and I finally found that it was due to the `remove_accidental_hits` switch. When this switch is set `True`, the `true_logit` of some sampled classes will be set `-inf` in function `_compute_sampled_logits` if they are accidentally the target, which causes `sampled_loss` to be `inf` since `label` is not zero because of label smoothing.
There are two possible solutions: 1) Considering that the total classes number is relative large, we can get approximate result even we turn off `remove_accidental_hits`. 2) Make sure that target will not be sampled.

BTW, I got similar result on my language model with `label_smoothing=0.2` after I turned off `remove_accidental_hits`.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
We're about to open-source our implementation in another repo. Will keep you posted.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 80 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
Please see the logic around [here](https://github.com/tensorflow/lingvo/blob/master/lingvo/core/layers.py#L2432).

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
